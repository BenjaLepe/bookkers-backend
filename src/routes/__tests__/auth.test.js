const supertest = require('supertest');
// const { format } = require('date-fns');
const app = require('../../app');

const request = supertest(app.callback());

describe('Auth API routes', () => {
  // crear usuario
  const userFields = {
    username: 'username_auth',
    first_name: 'name1',
    last_name: 'lastname1',
    email: 'email_auth@uc.cl',
    password: 'test123',
    description: 'soy una persona',
  };

  beforeAll(async () => {
    // sincronizar base de datos
    await app.context.orm.sequelize.sync({ force: true });

    // crear usuario en db
    await app.context.orm.User.create(userFields);
  });
  // cerrar la conexi칩n con sequelize
  afterAll(async () => {
    await app.context.orm.sequelize.close();
  });

  // 2. Inicio de sesi칩n de usuario
  describe('POST /auth', () => {
    let response;
    // usar variable request para hacer un post
    const authResponse = () => request
      .post('/auth')
      .set('Content-type', 'application/json')
      .send({ email: userFields.email, password: userFields.password });

    // la informaci칩n ingresada para registrarse es v치lida
    describe('user data is valid', () => {
      beforeAll(async () => {
        response = await authResponse();
      });

      test('responds with 200 status code', () => {
        expect(response.status).toBe(200);
      });

      test('responds with a json body type', () => {
        expect(response.type).toEqual('application/json');
      });

      test('body matches snapshot', () => {
        // expect(response.body).toMatchSnapshot();
        expect(response.body).toMatchSnapshot({ data: { access_token: expect.any(String) } });
      });
    });
  });
});
