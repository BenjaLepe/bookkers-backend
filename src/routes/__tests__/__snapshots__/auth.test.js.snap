// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth API routes POST /auth user data is valid body matches snapshot 1`] = `
Object {
  "code": 200,
  "data": Object {
    "access_token": Any<String>,
    "token_type": "Bearer",
    "user": Object {
      "description": "soy una persona",
      "email": "email_auth@uc.cl",
      "first_name": "name1",
      "id": 1,
      "is_admin": false,
      "last_name": "lastname1",
      "profile_picture": "https://www.pavilionweb.com/wp-content/uploads/2017/03/man-300x300.png",
      "username": "username_auth",
    },
  },
  "msg": "User logged successfully",
}
`;
